### ISSUE TITLE
Settings/Keys: New KeyPair Visualisation

### ISSUE THREAD

**Issue by vLooz** [lavab/web#759]
I got comments from some users, that the actual KeyPairs are not shown in a intuitive way. I've been working on a new UI to hopefully improve that. 

It's could be seen as a continuation of https://github.com/lavab/web/issues/326 and is obviously connected to https://github.com/lavab/web/issues/737

Whats different now?
The KeyPair are still shown in 2 states, collapsed and expanded view. Even though the collapsed view now shows more about the KeyPair logic - being made of Public and Private - and offers all the buttons a normal user needs right away. When the user wants to learn more about the keys, actually wants to see them or wants to do more fancy stuff like deleting etc. there is a new edit mode now for them. 
This edit mode is only accessible when the user has switched on advanced settings first. 

![2015-05-19_settings_keys new-01](https://cloud.githubusercontent.com/assets/11177812/7726326/79941bea-ff01-11e4-995e-680435326979.png)

![2015-05-19_settings_keys new-02](https://cloud.githubusercontent.com/assets/11177812/7726327/7d7301b8-ff01-11e4-892c-b9e4b6ed2385.png)

**Comment by vLooz** [comment]
![2015-05-19_settings_keys new-03](https://cloud.githubusercontent.com/assets/11177812/7797645/6dbe2ada-02ef-11e5-9d83-c7e8a3d8e95b.png)

**Comment by vLooz** [comment]
This mock shows the edit-view of an partially expanded KeyPair. Note the difference between the grey "header" in this mock and the grey "background" in the collapsed edit mode (2. mock). 

![2015-05-19_settings_keys new-04](https://cloud.githubusercontent.com/assets/11177812/7798233/946b5aba-02f7-11e5-96d2-224edea5edc5.png)

**Comment by vLooz** [comment]
Find the new icons (e.g. encrypted KeyPair etc.) here on Trello: https://trello.com/c/b9f8tDxd/46-updated-key-icons

**Comment by piggyslasher** [comment]
so we only show the delete icon if item is expanded. yes?

**Comment by vLooz** [comment]
No, sorry for the confusion. We can show it in both states:

![2015-05-19_settings_keys new-04-06](https://cloud.githubusercontent.com/assets/11177812/7800472/212bbfe4-0316-11e5-8151-8641497dee05.png)

**Comment by piggyslasher** [comment]
#guidelines from @vLooz 
![image](https://cloud.githubusercontent.com/assets/1991125/8003362/874a31a2-0ba9-11e5-9643-c898879723b7.png)

**Comment by joergG** [comment]
@piggyslasher - just cleaning up the waffle-board. this might be "key management rework" on trello? so implementation is finished?

**Comment by let4be** [comment]
hm I don't see this is in develop, yet

**Comment by joergG** [comment]
@piggyslasher - this would best be tested by fvl, right? can we also push it to staging?

**Comment by piggyslasher** [comment]
only thing that hasnt been merged. will merge later.

**Comment by joergG** [comment]
@piggyslasher - just tell us when it's done, so fvl can test this

**Comment by joergG** [comment]
@piggyslasher?

**Comment by piggyslasher** [comment]
@let4be is putting the finishing touches on this. I will then have another look after he's done to fix the encrypted/decrypted styling. I think that is all.

**Comment by let4be** [comment]
Should work, 
decrypt key popup probably needs some polishing @piggyslasher

as this is intermediate version I left buttons like "generate new key-pair" on their previous place
this will be addressed later

**Comment by vLooz** [comment]
Good to know you're still on it @let4be. I'll wait with my review then.

**Comment by let4be** [comment]
@vLooz I'm just saying we probably better do not introduce additional complexity here at the current stage(edit mode, pro mode, etc) and leave everything that possible to the next milestone(premium)

**Comment by vLooz** [comment]
I agree when it comes to introducing pro mode @let4be. 
However, from my pov the complexity already is there (additional key functions/buttons), just packaged in a way, that is not fitting the rest of the style. 
To increase our chances to convince investors, I think the UI should look as neat as possible. But I'm just the designer ;-)

**Comment by joergG** [comment]
@piggyslasher - so status is that you are styling this atm, right?

**Comment by piggyslasher** [comment]
Yes, I am getting everything done now. I needed to rewrite a lot of old code for this.

**Comment by piggyslasher** [comment]
More work on this. New guidelines.

![image](https://cloud.githubusercontent.com/assets/1991125/8908661/ecd6242a-34af-11e5-9d79-f09ea61f72ff.png)

**Comment by piggyslasher** [comment]
We currently don't have an *edit* mode @vLooz. So should I just ignore that for now? And re-open the issue when we have an edit mode?

**Comment by vLooz** [comment]
Why do we not have an edit mode @piggyslasher?
I told @let4be that we need it and I don't like to skip it a week ago.

**Comment by piggyslasher** [comment]
Sorry. Anyway. Have a look at http://lb.Ramanujam.am
Use your .com account to login. See the key section under settings.

On Wed, Jul 29, 2015, 17:09 Felix von Looz <notifications@github.com> wrote:

> Why do we not have an edit mode @piggyslasher
> <https://github.com/piggyslasher>?
> I told @let4be <https://github.com/let4be> that we need it and I don't
> like to skip it a week ago.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lavab/web/issues/759#issuecomment-125887465>.
>
-- 
Vik Ramanujam
vik@ramanuj.am (official)
piggyslasher@gmail.com (everything else)
+6016 656 4647
